File: websocket.java
import javax.websocket.*;
import java.net.URI;
import java.util.concurrent.TimeUnit;

@ClientEndpoint
public class WebSocketClient {

    private Session session;
    private final String uri = "ws://localhost:8080"; // Altere para o endereço do seu servidor WebSocket

    public WebSocketClient() {
        connect();
    }

    @OnOpen
    public void onOpen(Session session) {
        this.session = session;
        System.out.println("Conexão WebSocket estabelecida");
        sendMessage("Olá, servidor!");
    }

    @OnMessage
    public void onMessage(String message) {
        System.out.println("Mensagem do servidor: " + message);
    }

    @OnClose
    public void onClose(Session session, CloseReason closeReason) {
        System.out.println("Conexão WebSocket fechada. Motivo: " + closeReason);
        reconnect();
    }

    @OnError
    public void onError(Session session, Throwable throwable) {
        System.err.println("Erro no WebSocket: " + throwable.getMessage());
        reconnect();
    }

    private void connect() {
        try {
            WebSocketContainer container = ContainerProvider.getWebSocketContainer();
            container.connectToServer(this, URI.create(uri));
        } catch (Exception e) {
            e.printStackTrace();
            reconnect();
        }
    }

    private void reconnect() {
        try {
            TimeUnit.SECONDS.sleep(5); // Espera 5 segundos antes de tentar reconectar
            System.out.println("Tentando reconectar...");
            connect();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    public void sendMessage(String message) {
        try {
            session.getAsyncRemote().sendText(message);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static void main(String[] args) {
        new WebSocketClient();
    }
}

File: websocket.py
import asyncio
import websockets
import time

async def connect():
    uri = "ws://pornhub:8080"  
    while True:
        try:
            async with websockets.connect(uri) as websocket:
                print("hora da punheta")
                await websocket.send("denovo pai?")
                
                async for message in websocket:
                    print(f"Mensagem do servidor: {message}")
        except (websockets.ConnectionClosedError, websockets.ConnectionClosedOK) as e:
            print(f"Conexão perdida: {e}. Tentando reconectar...")
            await asyncio.sleep(5)  # Espera 5 segundos antes de tentar reconectar
        except Exception as e:
            print(f"Erro: {e}")
            await asyncio.sleep(5)  # Espera 5 segundos antes de tentar reconectar

async def main():
    await connect()

if __name__ == "__main__":
    asyncio.run(main())
